app/renderer/js/utils/link-util.js

```diff
@@ -21,6 +21,13 @@ class LinkUtil {
 
 		return (currentDomain === newDomain) && newUrl.includes('/#narrow');
 	}
+
+	isImage(url) {
+		const extension = url.substring(url.lastIndexOf('.') + 1).toLowerCase();
+		const imageExtensions = ['bmp', 'gif', 'jpg', 'jpeg', 'png', 'webp'];
```

Comment:
You didn't need to think about each and every possible image type here zulip only supports four types of image uploads see:

https://github.com/zulip/zulip/blob/5740af27d668b2a2f9942da170183779b7ae3b28/static/js/upload_widget.js#L7-L12

But this is probably fine :)

Reply:
Ah! I was searching that. I took from here https://github.com/zulip/zulip/blob/5740af27d668b2a2f9942da170183779b7ae3b28/static/js/markdown.js#L19 .
Can you verify which ones should be included?

Reply:
Those are for markdown, upload_widget only accepts these but it does not hurt to add them anyways so leave them there in fact i think i will be adding those in my PR so in case if we add some more image types in future we are safe.

Reply:
Sure that'd be good.

Reply:
Can you make this a regex check this right here is not some we would want: 
```javascript
const extension = url.substring(url.lastIndexOf('.') + 1).toLowerCase();
```

You can just do this instead:
```javascript
const isImageUrl = /\.(png|...)/;
return isImageUrl.test(url);
```

Reply:
One issue with this is that, if `.png` etc will come even in the middle of the url (which could be a one in a million case) then  it would return true.

Reply:
can you provide one example of such url?